Source/Fluids/WarpXFluidContainer.cpp

```diff
@@ -0,0 +1,194 @@
+
+        amrex::ParallelFor(tile_box, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+                           {
+#if defined(WARPX_DIM_3D)
+                               amrex::Real x = problo[0] + i * dx[0];
+                               amrex::Real y = problo[1] + j * dx[1];
+                               amrex::Real z = problo[2] + k * dx[2];
+#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+                amrex::Real x = problo[0] + i * dx[0];
+                amrex::Real y = 0.0_rt;
+                amrex::Real z = problo[1] + j * dx[1];
+#else
+                amrex::Real x = 0.0_rt;
+                amrex::Real y = 0.0_rt;
+                amrex::Real z = problo[0] + i * dx[0];
+#endif
+
+                               amrex::Real n = inj_rho->getDensity(x, y, z);
+                               auto u = inj_mom->getBulkMomentum(x, y, z);
+
+                               N_arr(i, j, k) = n;
+                               NUx_arr(i, j, k) = n * u.x;
+                               NUy_arr(i, j, k) = n * u.y;
+                               NUz_arr(i, j, k) = n * u.z; });
+    }
+
+    // Fill guard cells
+    const amrex::Periodicity &period = geom.periodicity();
+    FillBoundary(*N[lev], N[lev]->nGrowVect(), WarpX::do_single_precision_comms, period);
+    FillBoundary(*N[lev], NU[0][lev]->nGrowVect(), WarpX::do_single_precision_comms, period);
+    FillBoundary(*N[lev], NU[1][lev]->nGrowVect(), WarpX::do_single_precision_comms, period);
+    FillBoundary(*N[lev], NU[2][lev]->nGrowVect(), WarpX::do_single_precision_comms, period);
+}
+
+void WarpXFluidContainer::Evolve(int lev,
+                                 const amrex::MultiFab &Ex, const amrex::MultiFab &Ey, const amrex::MultiFab &Ez,
+                                 const amrex::MultiFab &Bx, const amrex::MultiFab &By, const amrex::MultiFab &Bz,
+                                 amrex::MultiFab &jx, amrex::MultiFab &jy, amrex::MultiFab &jz,
+                                 amrex::MultiFab *cjx, amrex::MultiFab *cjy, amrex::MultiFab *cjz,
+                                 amrex::MultiFab *rho, amrex::MultiFab *crho,
+                                 const amrex::MultiFab *cEx, const amrex::MultiFab *cEy, const amrex::MultiFab *cEz,
+                                 const amrex::MultiFab *cBx, const amrex::MultiFab *cBy, const amrex::MultiFab *cBz,
+                                 amrex::Real t, amrex::Real dt, DtType a_dt_type, bool skip_deposition)
+{
+    // Gather E&B fields to each node
+    // TODO
+
+    // Update N and NU at each node
+    // TODO
+
+    // Deposit J to the simulation mesh
+    if (!skip_deposition)
+    {
+        DepositCurrent(lev, jx, jy, jz);
+    }
+}
+
+void WarpXFluidContainer::DepositCurrent(int lev,
+                                         amrex::MultiFab &jx, amrex::MultiFab &jy, amrex::MultiFab &jz)
+{
+
+    const amrex::ParticleReal inv_clight_sq = 1.0_prt / PhysConst::c / PhysConst::c;
```

Comment:
```suggestion
    const amrex::Real inv_clight_sq = 1.0_prt / PhysConst::c / PhysConst::c;
```
Since we are not using particles here, I would recommend using `amrex::Real` everywhere. (`amrex::ParticleReal` is used when we want WarpX to use a different precision for particles and fields - e.g. single-precision for particles and double-precision for fields.)

Reply:
Fixed.